src/StoppingCondition.jl

```diff
@@ -0,0 +1,86 @@
+export AddStoppingCondition
+
+
+function AddStoppingCondition(p::DSProblem, c::T) where T <: AbstractStoppingCondition
+    push!(p.stoppingconditions, c)
+end
+
+CheckStoppingConditions(p::DSProblem) = CheckStoppingConditions(p, p.stoppingconditions)
```

Comment:
See my comment about function naming. These internal-only functions that just wrap around things should be prefixed by the underscore.